### PR TITLE
OSh/feature-TopicFiltering

### PR DESCRIPTION
**Pull request by voidshah**
Two filters (by 'title', 'description' and by tags ids) were created and ready for use in requests.

### CODE REVIEW

File: app/Http/Controllers/MessageController.php [hunk 1]
```diff
@@ -29,7 +29,7 @@ public function index($userId)
         $usersTo = User::whereIn('id', $usersToIds)->get();
 
         return $this->setStatusCode(200)->respond(
-            $messages->makeHidden('created_at', 'updated_at'),
+            $messages->makeHidden(['created_at', 'updated_at']),
```
Comment (TatsuUkraine) [inline app/Http/Controllers/MessageController.php]: @voidshah don't hide those fields, they can be useful for frontend

**Comment by Pozharskyi** [comment]
fix by merge with next

**Comment by voidshah** [comment]
May be this will be better?!

```
public function index(TopicRequest $request)
    {

        $topics = (new Topic())->newQuery();

        $topics = $this->filterByQuery($topics, $request);

        $topics = $topics->get();

        return $this->setStatusCode(200)->respond($topics);
    }
```

and filter

```
protected function filterByQuery(Builder $query, ApiRequest $request)
    {
        $searchStr = $request->get('query');
        if ($searchStr) {
            $query = $query->where('name','LIKE','%'.$searchStr.'%')
                ->orWhere('description','LIKE','%'.$searchStr.'%');
        }

        return $query;
    }
```

This will remove the code duplication

**Comment by binaryvo** [comment]
1. Move logic of filtering (now is in filterByQuery method of controller) to some method in appropriate model;
2. You can make some single protected function for retrieving data from request and filtering topics and just transfer $topics there

**Comment by voidshah** [comment]
refactored

**Comment by binaryvo** [comment]
I think, that better to move next code from constructor to some separated method:
$this->request = $request;
$this->searchStr = $request->get('query');
$tagIds = $request->get('tag_ids');
$this->tagIds = ($tagIds) ? explode(',', $tagIds) : [];

**Comment by voidshah** [comment]
Like that?

```
    public function __construct(TopicRequest $request)
    {
        $this->setData($request);
    }

    protected function setData(TopicRequest $request)
    {
        $this->request = $request;

        $this->searchStr = $request->get('query');
        $tagIds = $request->get('tag_ids');
        $this->tagIds = ($tagIds) ? explode(',', $tagIds) : [];
    }

```

**Comment by binaryvo** [comment]
Great! Please add some comments for method setFiltersData()

**Pull request merged**